include/fixed_containers/fixed_red_black_tree_view.hpp

```diff
@@ -0,0 +1,386 @@
+#pragma once
+
+#include "fixed_containers/fixed_red_black_tree.hpp"
+
+#include <cassert>
+#include <cstdint>
+#include <functional>
+#include <iterator>
+#include <numeric>
+#include <unordered_set>
+#include <utility>
+
+namespace fixed_containers
+{
+
+template <class M, class N>
+constexpr std::common_type_t<M, N> align_up(M m, N n)
+{
+    assert(n > 0);
+    return m + n - 1 - (m + n - 1) % n;
+}
+
+class FixedRedBlackTreeRawView
+{
+    using Compactness = fixed_red_black_tree_detail::RedBlackTreeNodeColorCompactness;
+    using CompactnessBacking = fixed_red_black_tree_detail::detail::RedBlackTreeNodeColorCompactnessBackingEnum;
+    using StorageType = fixed_red_black_tree_detail::RedBlackTreeStorageType;
+    using NodeIndex = fixed_red_black_tree_detail::NodeIndex;
+
+    static constexpr auto NULL_INDEX = fixed_red_black_tree_detail::NULL_INDEX;
+    
+public:
+    class Iterator
+    {
+    public:
+        using value_type = const std::byte*;
+        using difference_type = std::ptrdiff_t;
+        using pointer = value_type*;
+        using const_pointer = const value_type*;
+        using reference = value_type&;
+        using const_reference = const value_type&;
+        using iterator_category = std::forward_iterator_tag;
+
+        constexpr Iterator(const std::byte* ptr,
+                           std::size_t elem_size_bytes,
+                           std::size_t max_size_bytes,
+                           Compactness compactness,
+                           StorageType storage_type,
+                           bool end = false) noexcept
+          : base_(ptr)
+          , elem_size_bytes_(elem_size_bytes)
+          , max_size_bytes_(max_size_bytes)
+          , compactness_(compactness)
+          , storage_type_(storage_type)
+          , storage_elem_size_bytes_(storage_elem_size_bytes())
+          , index_(end ? NULL_INDEX : min_index())
+          , cur_pointer_(node_pointer(index_))
+        {}
```

Comment:
`clang-format` will fix a couple of violations in this file.

Reply:
c22ad6b